subsys/bootloader/Kconfig

```diff
@@ -116,23 +116,43 @@ config SB_DEBUG_SIGNATURE_PUBLIC_KEY_LAST
 	  first. This is meant to be used for testing looping through the
 	  public keys.
 
-endif # SECURE_BOOT
-
-menuconfig IS_SECURE_BOOTLOADER
-	bool "Current app is bootloader"
-	depends on SECURE_BOOT_VALIDATION
-	depends on SECURE_BOOT_STORAGE
-	select SW_VECTOR_RELAY if SOC_SERIES_NRF51X
-	select NRFX_NVMC
+config SB_MONOTONIC_COUNTER
+	bool "Enable HW monotonic version counter"
+	default y
 	help
-	  This option is set by the first stage bootloader app to include all
-	  files and set all the options required.
+	  An array of slots is used for one counter. One slot is used per
+	  counter update, and each new slot must have a larger value than the
+	  previous. Application versions are checked against the current largest
+	  counter before being booted.
+
+config SB_NUM_VER_COUNTER_SLOTS
+	int "Number of monotonic counter slots used for the firmware version."
+	default 240
+	range 1 300 if SOC_NRF5340_CPUAPP || SOC_NRF9160
+	range 1 1800 if SOC_SERIES_NRF52X
+	range 1 400 if SOC_SERIES_NRF51X
+	depends on SB_MONOTONIC_COUNTER
+	help
+	  The number of monotonic counter slots available for the counter.
+	  I.e. the number of times it can be updated.
+	  The slots are 16 bits each, and the slot array is preceded by a
+	  16 bit header. Because of this, the number of slots will be rounded
+	  up to the nearest odd number to ensure that header+slots is aligned on
+	  a 32 bit word.
```

Comment:
```suggestion
	  16-bit header. Because of this, the number of slots is rounded up to the
	  nearest odd number to ensure that the total size of header and slots is aligned on
	  a 32-bit word.
```

Reply:
This is no longer correct, I rewrote the text, and copied in your changes, see comment below.